setup.py

```diff
@@ -7,8 +7,15 @@
 
 # Determine if a base directory has been provided with the --basedir option
 in_tree = False
+# Add compiler flags if debug is set
+compile_args = []
 for arg in sys.argv:
-    if arg.startswith('--basedir='):
+    if arg.startswith('--debug'):
+        # Note from GCC manual:
+        #       If you use multiple -O options, with or without level numbers,
+        #       the last such option is the one that is effective.
+        compile_args.extend('-Wall -O0 -g'.split())
+    elif arg.startswith('--basedir='):
```

Comment:
This should probably stay as an `if` because we would like to allow both `--debug` and `--basedir`.